### DISCUSSION TITLE
support tutanota encryption

### DISCUSSION THREAD

**Discussion by bingoxo** [simple-login/app#643] [Feature Requests]
I love that you implemented PGP encryption , but since tutanota has their own encryption program (less adopted than pgp , but more private) , they are open source , maybe SL can work with it for e2e encryption ?

**Comment by ghost** [comment]
@nguyenkims i'm only ask, is this something the dev team thinks it possible to add in the near future or we have to wait more longer (only ask because since your service is my "front" mail) each year i decide if i rest on my current email service or if i switch (in this case for tutanota if you support it).

As dev i understand they are more shiny feature (i mean by that "will be used by more user at day one") just do not forget tutanota support :/

Thanks

**Reply by nguyenkims** [reply]
@BirdInFire hey we have been trying to get in touch with Tutanota team to integrate their encryption. As it isn't based on PGP, it's easier to have the Tutanota team giving a hand to support their encryption.

**Reply by ghost** [reply]
> @BirdInFire hey we have been trying to get in touch with Tutanota team to integrate their encryption. As it isn't based on PGP, it's easier to have the Tutanota team giving a hand to support their encryption.

Any news (or drop since proton deal ?)

**Reply by nguyenkims** [reply]
We don't hear back from Tutanota unfortunately. If you have a way to contact their team, please don't hesitate.

SimpleLogin joining Proton doesn't affect this in any way, SimpleLogin continues to be compatible with any email provider.

**Reply by ghost** [reply]
> We don't hear back from Tutanota unfortunately. If you have a way to contact their team, please don't hesitate.
> 
> SimpleLogin joining Proton doesn't affect this in any way, SimpleLogin continues to be compatible with any email provider.

have sended them a mail on the premium mail, and asked them to reach you on this post to explain how you can reach their dev team (or any team who have the power to help for this implementation) and at least if they can't answer on Git, tell me, like that i can tell you here.

will copy past the answer when (if) i receive one.

**Reply by ghost** [reply]
> We don't hear back from Tutanota unfortunately. If you have a way to contact their team, please don't hesitate.
> 
> SimpleLogin joining Proton doesn't affect this in any way, SimpleLogin continues to be compatible with any email provider.

they have responded : 

Hi there,
Thank you for your email. We've forwarded this to our development team.

Best regards,
Your Tutanota Team

Will post if i receive anything else.

**Comment by kevinlekiller** [comment]
This is not a good idea in my opinion, it's non standard so it's only useful for 1 specific email provider, it also means more code to maintain, and if Tutanota closes, the time and effort would have been wasted.

Also, Tutanota has wrote multiple times in the past they will support PGP in the future.

Here's a post from 9 days ago for example: https://teddit.net/r/tutanota/comments/uakiyq/what_does_tutanota_not_supporting_pgp_mean_in/i60nkuo

**Reply by ghost** [reply]
> This is not a good idea in my opinion, it's non standard so it's only useful for 1 specific email provider, it also means more code to maintain, and if Tutanota closes, the time and effort would have been wasted.
> 
> Also, Tutanota has wrote multiple times in the past they will support PGP in the future.
> 
> Here's a post from 9 days ago for example: https://teddit.net/r/tutanota/comments/uakiyq/what_does_tutanota_not_supporting_pgp_mean_in/i60nkuo

Tutanota is technicaly "the only one" truly E2E and the ONLY one who have postquantum cryptography functionnal (in beta and buggy), since all who use pgp can know : who send you a mail and the subjet in clear, and the contact email (in contact) and their name also are unencrypted, so yes they are drayback to tutanota, but say it's not a good idea is a bad idea.

Because if tuta/simplelogin appen, it mean proton/tutanota will arrive not long after, and it's always better to let the communication appen encrypted between services, than only use it unencrypted, becaus now two very "privacy" services are reduced to send plain text email between them.

So no having a module for each encrypted services is for me a plus for the future.

**Reply by ghost** [reply]
Totally agree.

**Comment by ghost** [comment]
I don't think tutanota is reliable enough. It is based in Germany, which is famous for its surveillance and also for its cooperation with the US. In addition, Germany is part of FVEY (https://en.wikipedia.org/wiki/Five_Eyes). Therefore, it is best to maintain the standard PGP. It is already a more widespread system and also more secure, since PGP has been tested for years.

**Reply by ghost** [reply]
> I don't think tutanota is reliable enough. It is based in Germany, which is famous for its surveillance and also for its cooperation with the US. In addition, Germany is part of FVEY (https://en.wikipedia.org/wiki/Five_Eyes). Therefore, it is best to maintain the standard PGP. It is already a more widespread system and also more secure, since PGP has been tested for years.

Ah yes this famous "standard" who leak nearly every metadata, sure ....
And us who use tutanota don't care about if other "think" it's reliable, we ask the feature and simple login dev, are not against, so if you don't like tuta don't use it but don't try to stop us to have better security.

Oh and spoiler alert E2E even in five eyes is secure....
It's the goal of E2E, we don't have to trust them, we just have to verify the sources.

But since you use PGP who leak the email / subject (and if you use proton (the name of your contact and their email)), zero trust isn't something you must know.

**Reply by ghost** [reply]
In that case, it is better to already use encryption similar to that used by ProtonMail. Regarding the address of your contact. I have clarified the information. In the case of Tuta, you can figure it out from the hash sum.
Well and I was talking about Tuta in general. In case of E2e it is indeed secure, but! There is one but. The encryption offered by Tuta hasn't been as time-tested as PGP. Well, in the case of receiving emails from third-party services other than Tuta, e2e is simply impossible (for example, from gmail to Tuta), and thus your emails are at risk due to the union of five eyes.

**Reply by ghost** [reply]
> In that case, it is better to already use encryption similar to that used by ProtonMail. Regarding the address of your contact. I have clarified the information. In the case of Tuta, you can figure it out from the hash sum. Well and I was talking about Tuta in general. In case of E2e it is indeed secure, but! There is one but. The encryption offered by Tuta hasn't been as time-tested as PGP. Well, in the case of receiving emails from third-party services other than Tuta, e2e is simply impossible (for example, from gmail to Tuta), and thus your emails are at risk due to the union of five eyes.

True AES & RSA isn't time tested at all : "Tutanota uses symmetric (AES 128) and asymmetric encryption (AES 128 / RSA 2048) to encrypt emails end-to-end. When both parties use Tutanota, all emails are automatically end-to-end encrypted (asymmetric encryption). For an encrypted email to an external recipient, a password for encrypting & decrypting the email (symmetric encryption) must be exchanged once." oh and they have PFS not like some basic PGP i know.

Stop bad mouthing them and check if they are secure before try push their user away ...

**Reply by ghost** [reply]
> In that case, it is better to already use encryption similar to that used by ProtonMail. Regarding the address of your contact. I have clarified the information. In the case of Tuta, you can figure it out from the hash sum. Well and I was talking about Tuta in general. In case of E2e it is indeed secure, but! There is one but. The encryption offered by Tuta hasn't been as time-tested as PGP. Well, in the case of receiving emails from third-party services other than Tuta, e2e is simply impossible (for example, from gmail to Tuta), and thus your emails are at risk due to the union of five eyes.

and the doc is so hard to find and verify they tell the truth a F12 on you browser is so hard, i've broke my finder ... :
https://tutanota.com/security/

**Reply by ghost** [reply]
`For an encrypted email to an external recipient, a password for encrypting & decrypting the email (symmetric encryption) must be exchanged once."`
You are talking about SENDING a letter. I am talking about receiving it.

`"Tutanota uses symmetric (AES 128) and asymmetric encryption (AES 128 / RSA 2048) to encrypt emails end-to-end`
Proton uses RSA 4096 and AES 256.

`Stop bad mouthing them and check if they are secure before try push their user away` 
I say what I see. Countries cooperating with the US cannot in principle be a safe solution for hosting servers.

`i've broke my finder`
I'm sorry to hear that.

**Reply by ghost** [reply]
> `For an encrypted email to an external recipient, a password for encrypting & decrypting the email (symmetric encryption) must be exchanged once."` You are talking about SENDING a letter. I am talking about receiving it.
> 
> `"Tutanota uses symmetric (AES 128) and asymmetric encryption (AES 128 / RSA 2048) to encrypt emails end-to-end` Proton uses RSA 4096 and AES 256.
> 
> `Stop bad mouthing them and check if they are secure before try push their user away` I say what I see. Countries cooperating with the US cannot in principle be a safe solution for hosting servers.
> 
> `i've broke my finder` I'm sorry to hear that.

Nope you have said : 

`I don't think tutanota is reliable enough. It is based in Germany, which is famous for its surveillance and also for its cooperation with the US. In addition, Germany is part of FVEY (https://en.wikipedia.org/wiki/Five_Eyes). Therefore, it is best to maintain the standard PGP. It is already a more widespread system and also more secure, since PGP has been tested for years.`

So first Zero trust do that we don't care witch country it's in
Second Also false that pgp is more secure (AES is more secure than pgp) is so false, because AES is (spoiler alert) more used than pgp.
